packages/@aws-cdk/aws-ec2/lib/volume.ts

```diff
@@ -210,4 +213,344 @@ export enum EbsDeviceVolumeType {
+   */
+  GENERAL_PURPOSE_SSD = GP2,
+
+  /**
+   * Highest-performance SSD volume for mission-critical low-latency or high-throughput workloads.
+   */
+  PROVISIONED_IOPS_SSD = IO1,
+
+  /**
+   * Low-cost HDD volume designed for frequently accessed, throughput-intensive workloads.
+   */
+  THROUGHPUT_OPTIMIZED_HDD = ST1,
+
+  /**
+   * Lowest cost HDD volume designed for less frequently accessed workloads.
+   */
+  COLD_HDD = SC1,
+
+  /**
+   * Magnetic volumes are backed by magnetic drives and are suited for workloads where data is accessed infrequently, and scenarios where low-cost
+   * storage for small volume sizes is important.
+   */
+  MAGNETIC = STANDARD,
+}
+
+/**
+ * An EBS Volume in AWS EC2.
+ */
+export interface IVolume extends IResource {
+  /**
+   * The EBS Volume's ID
+   *
+   * @attribute
+   */
+  readonly volumeId: string;
+
+  /**
+   * The availability zone that the EBS Volume is contained within (ex: us-west-2a)
+   */
+  readonly availabilityZone: string;
+
+  /**
+   * The customer-managed encryption key that is used to encrypt the Volume.
+   *
+   * @attribute
+   */
+  readonly encryptionKey?: IKey;
+
+  /**
+   * Grants permission to attach this Volume to an instance.
+   * @param grantee  the principal being granted permission.
+   * @param instance the instance to which permission is being granted to attach this
+   *                 volume to. If not specified, then permission is granted to attach
+   *                 to all instances in this account.
+   */
+  grantAttachVolume(grantee: IGrantable, instance?: IInstance): Grant;
+
+  /**
+   * Grants permission to detach this Volume from an instance
+   * @param grantee  the principal being granted permission.
+   * @param instance the instance to which permission is being granted to detach this
+   *                 volume from. If not specified, then permission is granted to attach
+   *                 to all instances in this account.
+   */
+  grantDetachVolume(grantee: IGrantable, instance?: IInstance): Grant;
+}
+
+/**
+ * Properties of an EBS Volume
+ */
+export interface VolumeProps {
+  /**
+   * The value of the physicalName property of this resource.
+   *
+   * @default The physical name will be allocated by CloudFormation at deployment time
+   */
+  readonly volumeName?: string;
+
+  /**
+   * The Availability Zone in which to create the volume.
+   */
+  readonly availabilityZone: string;
+
+  /**
+   * The size of the volume, in GiBs. You must specify either a snapshot ID or a volume size.
+   * See {@link https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ebs-volume-types.html#ebs-volume-characteristics|Volume Characteristics}
+   * for details on the allowable size for each type of volume.
+   *
+   * @default If you're creating the volume from a snapshot and don't specify a volume size, the default is the snapshot size.
+   */
+  readonly sizeGiB?: number;
```

Comment:
I think we have `cdk.Size` these days, so you can type it:

```
readonly size: Size;
```

And example usage is as:

```
size: Size.gibibytes(500)
```

Reply:
Oh, that's way better!